web/src/routes/(user)/photos/[[assetId=id]]/+page.ts

```diff
@@ -1,17 +1,49 @@
+import {
+  AssetGridOptionsValues,
+  createAssetGridOptionsFromArray,
+  QueryParameter,
+  type IQueryParameter,
+} from '$lib/constants';
+import { checkEnumInArray, parseNumberOrUndefined } from '$lib/utils';
 import { authenticate } from '$lib/utils/auth';
 import { getFormatter } from '$lib/utils/i18n';
-import { getAssetInfoFromParam } from '$lib/utils/navigation';
+import { getAssetInfoFromParam, isExternalUrl } from '$lib/utils/navigation';
 import type { PageLoad } from './$types';
 
-export const load = (async ({ params }) => {
+export const load = (async ({ params, url }) => {
   await authenticate();
   const asset = await getAssetInfoFromParam(params);
   const $t = await getFormatter();
+  const options: IQueryParameter = {};
+
+  const getPreviousRoute = url.searchParams.get(QueryParameter.PREVIOUS_ROUTE);
+  if (getPreviousRoute && !isExternalUrl(atob(getPreviousRoute))) {
+    options.previousRoute = atob(getPreviousRoute);
+  }
+
+  const assetGridOptionss = url.searchParams.get(QueryParameter.ASSET_GRID_OPTIONS)?.split(',');
+  if (assetGridOptionss && checkEnumInArray(assetGridOptionss, AssetGridOptionsValues)) {
+    options.assetGridOptions = createAssetGridOptionsFromArray(assetGridOptionss);
+  }
+
+  const x1 = parseNumberOrUndefined(url.searchParams.get(QueryParameter.COORDINATESX1));
+  const x2 = parseNumberOrUndefined(url.searchParams.get(QueryParameter.COORDINATESX2));
+  const y1 = parseNumberOrUndefined(url.searchParams.get(QueryParameter.COORDINATESY1));
+  const y2 = parseNumberOrUndefined(url.searchParams.get(QueryParameter.COORDINATESY2));
```

Comment:
Lodash omit by would probably look better

Reply:
Not sure to understand how omitBy can help for that